test/src/main/java/org/axonframework/test/fixture/AxonTestPhase.java

```diff
@@ -284,14 +244,36 @@ default When when(Consumer<AxonTestPhase.When> onWhen) {
      */
     interface When {
 
+        interface CommandWhen {
+
+            /**
+             * Transitions to the "then" phase to validate the results of the test. This method completes the "when"
+             * phase, committing any Unit of Work that was started during this phase.
+             *
+             * @return a {@link Then} instance that allows validating the test results.
+             */
+            Then.CommandThen then();
+        }
+
+        interface EventWhen {
+
+            /**
+             * Transitions to the "then" phase to validate the results of the test. This method completes the "when"
+             * phase, committing any Unit of Work that was started during this phase.
+             *
+             * @return a {@link Then} instance that allows validating the test results.
+             */
+            Then.EventThen then();
+        }
```

Comment:
The most important changes are here.
Based on when phase we allow different assertions in Then phase.
I think it will play nicely with subscriptions and queries tests.